### PR TITLE
CSPL-2046: downward merge develop branch to azureblob feature branch

### PR DESCRIPTION
**Pull request by vivekr-splunk**
1. merging develop branch
2. changed documents to point to v4
3. added space holder for 2.1 in changelog

### CODE REVIEW

**Review by jryb — APPROVED** [review]

File: pkg/splunk/enterprise/searchheadcluster_test.go [hunk 1]
```diff
@@ -1505,10 +1505,10 @@ func TestSearchHeadClusterWithReadyState(t *testing.T) {
 	// simulate create stateful set
 	c.Create(ctx, statefulset)
 
-	// simulate create clustermaster instance before reconcilation
-	c.Create(ctx, &searchheadcluster)
+	// simulate create clustermanager instance before reconcilation
```
Comment (kumarajeet) [inline pkg/splunk/enterprise/searchheadcluster_test.go:1508]: nit: clustermanager in comment

**Review by sgontla — APPROVED** [review]

**Review by kumarajeet — APPROVED** [review]

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Release Charts
+
+on:
+  push:
+    branches:
+      - master
```
Comment (gaurav-splunk) [inline .github/workflows/release.yml:6]: Just a note: we will need to change this once we rename to master.

File: bundle/metadata/annotations.yaml [hunk 3]
```diff
@@ -5,7 +5,7 @@ annotations:
   operators.operatorframework.io.bundle.metadata.v1: metadata/
   operators.operatorframework.io.bundle.package.v1: splunk-operator
   operators.operatorframework.io.bundle.channels.v1: alpha
-  operators.operatorframework.io.metrics.builder: operator-sdk-v1.18.1
+  operators.operatorframework.io.metrics.builder: operator-sdk-v1.15.0
```
Comment (gaurav-splunk) [inline bundle/metadata/annotations.yaml:8]: is this correct?

File: docs/ChangeLog.md [hunk 4]
```diff
@@ -1,9 +1,20 @@
 # Splunk Operator for Kubernetes Change Log
 
+## 2.0.1 (TODO)
```
Comment (gaurav-splunk) [inline docs/ChangeLog.md:3]: Should be 2.1.0

File: .github/workflows/build-test-push-workflow.yml [hunk 5]
```diff
@@ -57,6 +55,7 @@ jobs:
       SPLUNK_OPERATOR_IMAGE_NAME: splunk/splunk-operator
       ECR_REPOSITORY: ${{ secrets.ECR_REPOSITORY }}
       S3_REGION: ${{ secrets.AWS_DEFAULT_REGION }}
+      DEPLOYMENT_TYPE: helm
```
[the remark this answers is not in the document]
Comment (pdhanoyasplunk) [reply .github/workflows/build-test-push-workflow.yml:58]: Its hardcoded on develop.

File: bundle/metadata/annotations.yaml (hunk 3 above)
Comment (pdhanoyasplunk) [reply bundle/metadata/annotations.yaml:8]: Automerged. I will change the value.

File: docs/ChangeLog.md (hunk 4 above)
Comment (pdhanoyasplunk) [reply docs/ChangeLog.md:3]: Automerged. I will change the value.

File: .github/workflows/build-test-push-workflow.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply .github/workflows/build-test-push-workflow.yml:58]: @pdhanoyasplunk fixing this issue on `develop` so once he is done, we should be able to merge that to feature branch. so for now it should be ok.

File: docs/ChangeLog.md (hunk 4 above)
Comment (vivekr-splunk) [reply docs/ChangeLog.md:3]: my bad I will fix it

File: bundle/metadata/annotations.yaml (hunk 3 above)
Comment (vivekr-splunk) [reply bundle/metadata/annotations.yaml]: this will be regenerated with different PR to the develop once all the develop changes are done from 2.1.0

**Review by gaurav-splunk — APPROVED** [review]

**Pull request merged**